helix-core/src/match_brackets.rs

```diff
@@ -70,10 +72,67 @@ fn find_pair(syntax: &Syntax, doc: &Rope, pos: usize, traverse_parents: bool) ->
     }
 }
 
+// Returns the position of the matching bracket under cursor, the search
+// is limited to the current line. This function works on plain text, it
+// ignores tree-sitter grammar.
+//
+// If the cursor is on the opening bracket, the position of
+// the closing bracket is returned. If the cursor on the closing
+// bracket, the position of the opening bracket is returned.
+//
+// If the cursor is not on a bracket, `None` is returned.
+//
+// If no matching bracket is found, `None` is returned.
+#[must_use]
+pub fn find_matching_bracket_current_line_plaintext(
+    doc: &Rope,
+    cursor_pos: usize,
+) -> Option<usize> {
+    // Don't do anything when the cursor is not on top of a bracket.
+    let bracket = doc.char(cursor_pos);
+    if !is_valid_bracket(bracket) {
+        return None;
+    }
+
+    // Determine the direction of the matching.
+    let is_fwd = is_forward_bracket(bracket);
+    let line = doc.byte_to_line(cursor_pos);
```

Comment:
positions are stored as char (utf-32) instead of byte (utf-8) positions in helix so `char_to_line` should be used here. 

That being said I don't actually see the point of limiting the matching to the current line. You could just search `cursor_pos + 1..` and `..cursor_pos`.rev() instead  that seems like it needs less code while being more useful.

Why did you introduce the limitation of limiting the matching to the current line?

Reply:
I just wanted to keep things simple and avoid scanning the entire document, seems like potentially expensive, so by constraining to only a line it would ensure there's no performance hit, while providing some usefulness already. I've removed it now.